### PR TITLE
dt-multiplelinearregression

### PR DESCRIPTION
**Pull request by Mohamadkhalil91**

### CODE REVIEW

**Comment by attila-balint-kul** [comment]
I would first start with a simple multivariate regression where we can use multiple horizons of lags similar to sf-naive-seasonal

This model takes 1 parameter, the season length such as 1D 7D etc. This way we can reuse the same model with different seasonalities.

File: models/dt-multiplelinearregression/src/main.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from enfobench.evaluation.metrics import mean_absolute_error, mean_bias_error,root_mean_squared_error
+from enfobench import AuthorInfo, ModelInfo, ForecasterType
+from enfobench.evaluation.server import server_factory
+from enfobench.evaluation.utils import create_forecast_index, periods_in_duration
+from enfobench.dataset.utils import create_perfect_forecasts_from_covariates
+
+class MultipleLinearRegressionDarts:
+
+    def __init__(self, seasonality: str):
+        self.seasonality = seasonality()
```
Comment (attila-balint-kul) [inline models/dt-multiplelinearregression/src/main.py]: You are calling a string parameter.

```python
def __init__(self, seasonality: str):
        self.seasonality = seasonality
```

Comment (attila-balint-kul) [reply models/dt-multiplelinearregression/src/main.py]: maybe add .upper() to the seasonality like this

```python
def __init__(self, seasonality: str):
        self.seasonality = seasonality.upper()
```

File: models/dt-multiplelinearregression/src/main.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        horizon: int,
+        history: pd.DataFrame,
+        past_covariates: pd.DataFrame  | None = None,
+        future_covariates: pd.DataFrame | None = None,
+        **kwargs
+        
+    ) -> pd.DataFrame:
+        
+        periods = periods_in_duration(history.index, duration=self.seasonality)
+        model=RegressionModel(lags= periods, output_chunk_length=horizon,model=LinearRegression())
```
Comment (attila-balint-kul) [inline models/dt-multiplelinearregression/src/main.py]: this way you only use a single lag of 1D or 7D not all available lags until then.
I think what you ment was `lags=list(range(periods))`

File: models/dt-multiplelinearregression/src/main.py (hunk 1 above)
Comment (Mohamadkhalil91) [reply models/dt-multiplelinearregression/src/main.py]: You are calling a string parameter.

How should i define the seasonality?

Comment (attila-balint-kul) [reply models/dt-multiplelinearregression/src/main.py]: I am using pandas period strings to define seasonality such as `1D` for 1 day etc.

Comment (Mohamadkhalil91) [reply models/dt-multiplelinearregression/src/main.py]: I have fixed this, and the code should be working now.

File: models/dt-multiplelinearregression/src/main.py (hunk 2 above)
Comment (Mohamadkhalil91) [reply models/dt-multiplelinearregression/src/main.py]: that should works now.

**Review by attila-balint-kul — APPROVED** [review]
Looks good now, I will do some cleanup and formatting.

**Pull request merged**